test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommand/FallbackFolderRestoreTests.cs

```diff
@@ -450,7 +451,15 @@ public async Task FallbackFolderRestore_VerifyMissingFallbackFolderFailsAsync()
 
                 // Act & Assert
                 var command = new RestoreCommand(request);
-                await Assert.ThrowsAsync<FatalProtocolException>(async () => await command.ExecuteAsync());
+                var result = await command.ExecuteAsync();
+                await result.CommitAsync(logger, CancellationToken.None);
+
+                Assert.False(result.Success);
+                // ideally, this error wouldn't repeat itself in the same project.
+                Assert.Equal(2, logger.ErrorMessages.Count());
```

Comment:
Would it be better to use a `Contains` check against the specific `NU` codes to make this less fragile?

Reply:
Yes, this string check isn't gorgeous...but RestoreCollectingLogger doesn't seem to collect NUXXXX info, if i recall.
I judged it not worth doing at this point.